packages/enhanced/src/lib/container/ContainerPlugin.ts

```diff
@@ -210,32 +214,41 @@ class ContainerPlugin {
           },
           (error: WebpackError | null | undefined) => {
             if (error) return callback(error);
-            if (hasSingleRuntimeChunk) {
-              // Add to single runtime chunk as well.
-              // Allows for singleton runtime graph with all needed runtime modules for federation
-              addEntryToSingleRuntimeChunk();
-            } else {
-              callback();
-            }
+            callback();
           },
         );
+      },
+    );
 
-        // Function to add entry for undefined runtime
-        const addEntryToSingleRuntimeChunk = () => {
-          compilation.addEntry(
-            compilation.options.context || '',
-            dep,
-            {
-              name: name ? name + '_partial' : undefined, // give unique name name
-              runtime: undefined,
-              library,
-            },
-            (error: WebpackError | null | undefined) => {
-              if (error) return callback(error);
-              callback();
-            },
-          );
-        };
+    compiler.hooks.make.tapAsync(
```

Comment:
split this into two hooks, the original container hook to create remoteEntry, then another sync hook to add an additional one for runtime single runtime chunk to have runtime requirements